apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -52,6 +53,24 @@ class PageTemplateModal extends Component {
 		)
 	);
 
+	getBlocksForPreview = memoize( previewedTemplate => {
+		const blocks = this.getBlocksByTemplateSlug( previewedTemplate );
+
+		// Modify the existing blocks returning new block object references.
+		return modifyBlocks( blocks, function modifyBlocksForPreview( block ) {
+			// `jetpack/contact-form` has a placeholder to configure form settings
+			// we need to disable this to show the full form in the preview
+			if (
+				'jetpack/contact-form' === block.name &&
+				undefined !== block.attributes.hasFormSettingsSet
+			) {
+				block.attributes.hasFormSettingsSet = true;
+			}
```

Comment:
I'm thinking we could remove this and replace with a filter hook. Then we could simply register filters against the hook based on block name. That way we wouldn't need to store all the individual block enhancements within the component. Rather there would be an individual filter per modification. This would make it clearer which blocks were modifying the preview output and for what reasons. 

If we don't do this then there's the potential for multiple `if` or `switch` statements growing within this section.

Might be a refactor too far. Let me know what you think.

Reply:
I'm fine with `switch` or `if`. Let build more infra only once we have the need